LICENSE.txt

```diff
@@ -1,84 +1,188 @@
+  the license. Code from MemoryModule can be found in
+  "racket/src/start/MemoryModule.{c,h}".
+
+The following are used only in Racket BC executables:
+
+* libscheme
+  Copyright © 1994 Brent Benson
+  All rights reserved.
+  See the file "racket/src/LICENSE-libscheme.txt" for the full text of
+  the libscheme license. Code from libscheme can be found in
+  "racket/src/bc/src".
+
+* GNU Lightning
+  Copyright © 1994, 1995, 1996, 1999, 2000, 2001, 2002, 2011 Free
+  Software Foundation, Inc.
+  GNU Lightning is distributed under the GNU Lesser General Public
+  License, version 3, or (at your option) any later version.
+  A fork of GNU Lightning can be found in "racket/src/bc/src/lightning".
+
+* GNU MP Library
+  Copyright © 1991, 1992, 1993, 1994, 1996, 1999, 2000, 2007 Free
+  Software Foundation, Inc.
+  GMP is distributed under the GNU Lesser General Public License,
+  version 3, or (at your option) any later version.
+  Code from GMP can be found in "racket/src/bc/src/gmp".
+
+* libunwind
+  Copyright © 2003–2005 Hewlett-Packard Development Company, L.P.
+  libunwind is distributed under the MIT license.
+  Code from libunwind can be found in "racket/src/bc/src/unwind".
+
+* libffi
+  Copyright © 1996–2019 Anthony Green, Red Hat, Inc and others.
+  libffi is distributed under the MIT license.
+  A copy of libffi can be found in "racket/src/bc/foreign/libffi".
+
+* random.inc
+  Based on "random.c" from FreeBSD 2.2.
+  Copyright © 1983, 1993 The Regents of the University of California.
+  This code is distributed under a three-clause BSD license.
+  Code from FreeBSD 2.2 can be found in "racket/src/bc/src/random.inc".
+
+In addition to the Racket runtime system, the default mode of building
+Racket will install some packages, which are distributed under their own
+licenses.
+
+The Racket code for all packages in the main Racket distribution is
+primarily under the MIT license and the Apache License, version 2.0, at
+your option, except that some packages contain code under other
+permissive licenses, and the "string-constants-lib-lgpl" package (which
+provides Spanish, Korean, and Dutch GUI translations) is distributed
+under the GNU Lesser General Public License, version 2.0, or (at your
+option) any later version.
+
+Some components of Racket are implemented using additional shared
+libraries. For some platforms and installation modes (especially for
+Windows and Mac OS), these libraries are distributed in
+platform-specific Racket packages; in other cases, they may be included
+with the operating system or installable via the system’s package
+manager. In all cases, these libraries are dynamically loaded and can be
+replaced with modified variants by users. In particular:
+
+* Even the “minimal Racket” distribution may include shared libraries
+  for OpenSSL, SQLite, and/or Editline, depending on the platform.
+
+  The Windows distribution also includes an iconv
+  implementation—currently GNU libiconv, which is distributed under the
```

Comment:
The [documentation currently](https://docs.racket-lang.org/reference/bytestrings.html#%28part._.Bytes_to_.Bytes_.Encoding_.Conversion%29) avoids committing to any particular set of conversions being available, which seemed wise, especially in light of e.g. https://github.com/racket/racket/issues/4277. GNU libiconv, on the other hand, does have a documented list of encodings and that conversion is supported for all pairs of encodings. I think we should list the license of what we actually distribute, but I tried to making a new commitment to support all of GNU libiconv's conversions.

I don't feel especially strongly about the wording, though.

Reply:
I agree about the conversion documentation, but I think here we should just describe the things we distribute rather than something about future possibilities.

Reply:
I've removed "currently" in the latest series. I haven't added any language about libraries potentially changing.